automation_tools/satellite6/upgrade/template.py

```
@@ -0,0 +1,121 @@
+        if check.return_code == 0:
+            print "running capsule sync"
+    except Exception as ex:
+        print ex
+        sys.exit(1)
+
+
+# Check if ntpd is running
+def check_ntpd():
+    try:
+        ntpd_check = run("service ntpd status", warn_only=True)
+        if ntpd_check.return_code > 0:
+            run("service ntpd start")
+            run("chkconfig ntpd on")
+    except Exception as ex:
+        print ex
+        sys.exit(1)
+
+
+# Check for Vm's and create their templates
+def create_template(host, cluster, new_template, storage):
+    """ Create template from vmName """
+    try:
+        get_client = get_rhevm_client()
+        storage_domain = get_client.storagedomains.get(name=storage)
+        size = storage_domain.get_available() / 1024 / 1024 / 1024
+        vm = get_rhevm_client().vms.get(host)
+        if size > 300 and vm:
+            try:
+                vm.stop()
+                wait_till_rhev_instance_status(host, 'down', 5)
+                print 'Waiting for VM to reach Down status'
+                get_client.templates.add(
+                    params.Template(name=new_template,
+                                    vm=get_client.vms.get(host),
+                                    cluster=get_client.clusters.get(cluster)))
+                while get_client.vms.get(host).status.state != 'down':
+                    sleep(1)
+                    "Template creation in Progress"
+            except Exception as ex:
+                get_client.disconnect()
+                print 'Failed to Create Template from VM:\n%s' % str(ex)
+        else:
+            get_client.disconnect()
+            print "Low Storage cannot proceed or VM not found"
+            sys.exit(1)
+    except Exception as ex:
+        get_client.disconnect()
+        print ex
+        sys.exit(1)
+
+
+# Fabric task
+def validate_and_create():
+    sat_host = os.environ.get('RHEV_SAT_HOST')
+    cap_host = os.environ.get('RHEV_CAP_HOST')
```

Is is ok if those env vars are not defined? I think the code should check and fail if they are not provided since the remaining of the code is relying on those being defined.